src/amr/messengers/hybrid_hybrid_messenger_strategy.hpp

```diff
@@ -733,228 +772,178 @@ namespace amr
         }
 
 
-        void debug_print(VecFieldT const& B, GridLayoutT const& layout, int loc, int ix, int iy,
-                         std::string const& aftbef)
-        {
-            auto& Bx       = B(core::Component::X);
-            auto& By       = B(core::Component::Y);
-            auto& Bz       = B(core::Component::Z);
-            auto const& dx = layout.meshSize()[0];
-            auto const& dy = layout.meshSize()[1];
 
-            if (loc == 3) // w hi, y hi
+
+        void magneticRegriding_(std::shared_ptr<SAMRAI::hier::PatchHierarchy> const& hierarchy,
```

Comment:
I think at this end of this function we should run a function that checks that nowhere there is a NaN on any component of the magnetic field left.
This function could be ran only in certain debug mode we have..